zebra-state/src/service/finalized_state/zebra_db/block/tests/snapshots/orchard_trees@mainnet_2.snap

```diff
@@ -13,9 +13,7 @@ expression: stored_orchard_trees
     inner: Frontier(
       frontier: None,
     ),
-    cached_root: Some(Root(Base(
-      bytes: (174, 41, 53, 241, 223, 216, 162, 74, 237, 124, 112, 223, 125, 227, 166, 104, 235, 122, 73, 177, 49, 152, 128, 221, 226, 187, 217, 3, 26, 229, 216, 47),
-    ))),
+    cached_root: None,
```

Comment:
It might reduce CPU usage if we didn't calculate the cached root for block 2. Is that calculation happening in the state code or snapshot test code?

(It's not being stored in the raw data, so it's not a disk usage issue, and the tree is the same, so it's not a consensus issue.)

Reply:
Just to be clear, I'm not concerned about CPU usage in tests!

Reply:
I think this is ok, it's either due to the test, or due to a state API where the root is used.

Reply:
> Is that calculation happening in the state production code, or snapshot test code?

We calculate the roots of each tree for each block, including genesis, in the state. We also use the cached roots multiple times in the state. The main branch currently doesn't store the roots for the genesis block but does store them for each subsequent block.

This PR doesn't store the roots (more precisely, the trees) if they match the previous ones, but the state still calculates them and uses them. However, the root for height two in `orchard_trees@mainnet_2.snap` shows up because it's calculated in the test code, not in the state, as you pointed out! Huge thanks for that.

So the answer to the question is that the calculation happens in both the state production code and snapshot test code. Since this PR doesn't currently store effectively any Sapling and Orchard roots in the database until their activation, the state would need to recompute the genesis root for each block until their first change. This would be a performance regression for the initial sync, and we should store the genesis roots in the database, as you also pointed out!